app/relationreferencefeaturesmodel.cpp

```diff
@@ -18,35 +18,50 @@ RelationReferenceFeaturesModel::RelationReferenceFeaturesModel( QObject *parent
 
 RelationReferenceFeaturesModel::~RelationReferenceFeaturesModel() = default;
 
-QVariant RelationReferenceFeaturesModel::foreignKeyFromAttribute( FeaturesModel::ModelRoles fromAttribute, const QVariant &attributeValue )
+QVariant RelationReferenceFeaturesModel::foreignKeyFromReferencedFeatureId( QgsFeatureId fid )
 {
-  if ( mPrimaryKeyField.isEmpty() )
+  if ( !mRelation.isValid() )
     return QVariant();
```

Comment:
```suggestion
    return {};
```
I know you didn't change it, but we should when we are dealing with it anyway